### PR TITLE
[release-v1.4] Add pause/resume lifecyecle to ordered consumer (#2438)

### PR DESCRIPTION
**Pull request by pierDipi**
Cherry-pick of https://github.com/knative-sandbox/eventing-kafka-broker/commit/a370abc70df0dd788b926adfb587111169c2dc14

### CODE REVIEW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java [hunk 1]
```diff
@@ -163,12 +164,32 @@ private synchronized OrderedAsyncExecutor executorFor(final TopicPartition topic
...
       }
     }
-    return this.recordDispatcherExecutors.size() == 0;
+
+    if (!toPause.isEmpty()) {
+      this.consumer.pause(toPause);
+    }
+    if (!toResume.isEmpty()) {
+      this.consumer.resume(toResume);
+    }
```
Comment (matzew) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java:191]: Didn't Clement mention something around pause/replay issues on vertx Kafka, @pierDipi ?

**Review by matzew — APPROVED** [review]
/lgtm
/approve
(regardless of the comment, good to get this branch aligned with the upstream main)

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java (hunk 1 above)
Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java:191]: Yes, but this happens only for some calls https://github.com/vert-x3/vertx-kafka-client/issues/53

**Pull request merged**